### DISCUSSION TITLE
Tips, Tricks, SMD Soldering

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16690] [General]
I purchased an SMD rework station for soldering smaller points. It's got both controllable soldering iron and hot air.
Does anyone have any tips and tricks on how I can best tap into GND/TX/RX on tiny wireless boards. 
I've got soldering experience but these days, I need magnifying glasses too and don't want to ruin the boards.
I'm not getting into development, I just want a way to quickly tap into the above leads when tuya-convert won't work for example.

How do you tap onto those leads? Do you solder on a pin, a wire, something else? It's only for temp setup, nothing else.

**Comment by sfromis** [comment]
What I do is to put fresh solder (to have some flux) on both the wire and the point I want to connect to. After that, it should be quick to fuse them with only short time heat, of course while holding the wire still. And I do have a good magnifier lamp to help my eyesight :eyeglasses: 

It gets hard only when it is **really** tiny, like 8 pins along each edge of a 5x5 mm chip. I've refrained from that :wink: 

BTW. One of my habits, when feasible, is to make a permanent setup with wires to a connector (like a female jumper connector), to make it easy next time. The I label the connector with what pins are where.

**Reply by dac2020** [reply]
Thanks for your tips.

Someone mentioned a pitch of 2.00. Does that mean if I get a header with pins, I could lay that across the pins I want to solder, solder those on and now have a permanent connector. Maybe I need to find some little plastic headers with pins I can remove and leave as needed.

**Reply by sfromis** [reply]
Yes, pitch 2.0 mm means that each position is two mm apart. If you get a header with 2.0 mm pitch (2.54 mm is much more common), you could solder it in place, possibly after bending a bit to make things snug after soldering. Still depends on whether the device laves room for such tricks, this is quite often not the case.

What I've sometimes done, if it fit the board layout, has been to take a 2.54 mm header row with long bendable pins (taken from the pins delivered with a D1 Mini), and bend and/or snip these to match the points I wanted to connect to. The image on the template for [Xiaomi Desk Lamp Pro](https://templates.blakadder.com/xiaomi_MJTD02YL.html) is from me.

For those common dupont 2.54 mm headers, you can also buy empty shells, and even the metal part to crimp onto wires, but it is easier to remove the 1-position shell from common dupont wires, and insert into a wider block. Or simply glue/tape those single-position ones to a wider block.

**Reply by barbudor** [reply]
The best jig for programming an ESP12 on board would be to 3D print a jig that use pogo-pins. Pogo-pins are pins with a contact end mounted on a spring which allow to press multiple of them gently on the device and make contact even if they are not fully vertically aligned.
Exemple of 3D printed jig : https://www.thingiverse.com/thing:3382207
Pogo pins : https://www.aliexpress.com/wholesale?SearchText=pogo%20pin

Now you can do something similar by using headers at 2.0mm soldered on a 2.0 proto board
headers 2.0 : https://www.aliexpress.com/item/4000383643368.html (like the 17 or 19mm long ones)
2.0 proto board (hard to find) : https://www.aliexpress.com/item/1485957459.html
Cut 1 pieces of the PCB slighly larger than the ESP
solder 2 rows of headers on the PCB so that they fit on the ESP12 module. solder the short side on the PCB. The long side will press on the ESP12 contacts
Insure to have a very flat result to fit well on the ESP
It may be usefull to cut un-used pins to get better press-fit on the contacts
difficult to explain in words but I don't have any photos to show

**Reply by dac2020** [reply]
Thanks for all of the tips on this. Sorry I forgot to acknowledge reading the replies.
